contracts/feature-tests/composability/forwarder-queue/src/forwarder_queue.rs

```diff
@@ -64,30 +69,54 @@ pub trait ForwarderQueue {
             &esdt_transfers_multi.into_multi_value(),
         );
 
+        // Should not reach here
         if max_call_depth == 0 {
             return;
         }
 
         while let Some(node) = self.queued_calls().pop_front() {
-            let call = node.into_value();
-            let contract_call = self
-                .self_proxy(call.to)
-                .forward_queued_calls(max_call_depth - 1)
-                .with_egld_or_single_esdt_transfer((
-                    call.payment_token,
+            let call = node.clone().into_value();
+
+            // Got to the leaf -> call Vault
+            if max_call_depth == 1 {
+                let contract_call = ContractCallWithEgldOrSingleEsdt::<Self::Api, ()>::new(
+                    call.to.clone(),
+                    call.endpoint_name.clone(),
+                    call.payment_token.clone(),
                     call.payment_nonce,
-                    call.payment_amount,
-                ));
-            match call.call_type {
-                QueuedCallType::Sync => {
-                    contract_call.execute_on_dest_context::<()>();
-                },
-                QueuedCallType::LegacyAsync => {
-                    contract_call.async_call().call_and_exit();
-                },
-                QueuedCallType::TransferExecute => {
-                    contract_call.transfer_execute();
-                },
+                    call.payment_amount.clone(),
+                );
+                match call.call_type {
+                    QueuedCallType::Sync => {
+                        contract_call.execute_on_dest_context::<()>();
+                    },
+                    QueuedCallType::LegacyAsync => {
+                        contract_call.async_call().call_and_exit();
+                    },
+                    QueuedCallType::TransferExecute => {
+                        contract_call.transfer_execute();
+                    },
+                }
+            } else {
+                let contract_call = self
```

Comment:
There is no need to work with the proxy, leave this else out and just use the calls above.

We will program it from the data we provide to it, there is no need to for this contract to know about the vault. We pass it the endpoint to call, so we can choose to make it call endpoints from the vault, from itself, or from other contracts.